File: frontend/src/logger.js
const debug = import.meta.env.VITE_LOG_LEVEL === 'debug'

const logger = {
  debug: (...args) => { if (debug) console.debug('[HomeScan DEBUG]', ...args) },
  info:  (...args) => console.info('[HomeScan]', ...args),
  warn:  (...args) => console.warn('[HomeScan WARN]', ...args),
  error: (...args) => console.error('[HomeScan ERROR]', ...args),
}

export default logger

File: backend/src/logger.py
import logging
import os

_level_map = {
    "debug": logging.DEBUG,
    "info": logging.INFO,
    "warning": logging.WARNING,
    "error": logging.ERROR,
}

_level = _level_map.get(os.environ.get("LOGLEVEL", "info").lower(), logging.INFO)

logging.basicConfig(
    level=_level,
    format="%(asctime)s [%(levelname)s] %(name)s: %(message)s",
    datefmt="%Y-%m-%d %H:%M:%S",
)


def get_logger(name: str) -> logging.Logger:
    logger = logging.getLogger(name)
    logger.setLevel(_level)
    return logger
